components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java

```diff
@@ -685,20 +679,58 @@ private static void copyFolder(File sourceFolder, File destinationFolder) throws
      * Creates a new folder if not exists
      *
      * @param path folder path
-     * @return File object for the created folder
-     */
-    private static File createFolderIfNotExist(String path) {
-        File folder = new File(path);
-        if (!folder.exists() && !folder.isDirectory()) {
-            boolean created = folder.mkdir();
+     * @param overwrite if `true` existing directory will be removed
+     *                  and new directory will be created in {@code path}.
+     * @return created directory
+     * @throws IOException Failed delete or create the directory in {@code path}
+     */
+    public static File createDirectory(String path, boolean overwrite) throws IOException {
+        File dir = new File(path);
+        if (overwrite && dir.exists() && dir.isDirectory()) {
+            FileUtils.deleteDirectory(dir);
+        }
+
+        if (!dir.exists() && !dir.isDirectory()) {
+            boolean created = dir.mkdir();
             if (created) {
                 logger.trace("Directory: {} created. ", path);
             } else {
-                logger.error("Failed to create directory: {} ", path);
-                throw new CLIInternalException("Error occurred while setting up the workspace structure");
+                throw new CLIInternalException("Failed to create directory: " + path);
             }
         }
-        return folder;
+
+        return dir;
+    }
+
+    /**
+     * Create new file in a given location.
+     *
+     * @param path location of the new file.
+     * @param overwrite if `true` existing file with the same name will be replaced.
+     * @param fileName name of the new file.
+     * @return created file
+     * @throws IOException Failed delete or create the file in {@code path}
+     */
+    public static File createFile(String path, String fileName, boolean overwrite) throws IOException {
+        String filePath = path + File.separator + fileName;
+        File file = new File(filePath);
+        if (overwrite && file.exists() && file.isFile()) {
+            boolean isDeleted = file.delete();
+            if (!isDeleted) {
+                throw new CLIInternalException("Failed to overwrite file: " + filePath);
```

Comment:
Isn't it better to log these messages. I think we log only the e.message in the main function